src/data/abilities/ab-attrs/stab-boost-ab-attr.ts

```diff
@@ -51,7 +44,11 @@ export class StabBoostAbAttr extends AbAttr {
       // If the move type is one of the pokemon's original types then the STAB multiplier is increased by 0.5
       stabMultiplier.value += 0.5;
     }
+  }
 
-    return initialStabMultiplier !== stabMultiplier.value;
+  public override canApply(...[pokemon, , move]: Parameters<this["apply"]>): boolean {
+    return (
+      pokemon.getMoveType(move) !== ElementalType.STELLAR && pokemon.getTypes(true).includes(pokemon.getMoveType(move))
+    );
```

Comment:
1 less function call.
```suggestion
    const moveType = pokemon.getMoveType(move);
    return moveType !== ElementalType.STELLAR && pokemon.getTypes(true).includes(moveType);
```